### PR TITLE
ISSUE-2084: Fixing the masking and  node id for scan results

### PR DESCRIPTION
**Pull request by varunsharma0286**
Fixes #
https://github.com/deepfence/enterprise-roadmap/issues/2084

Changes proposed in this pull request:
- Changing the node id for the scan results to remove registry id and docker image name
- Added the "MASKED" relationship between the Image Stub and scan results to support Image level masking(mask for all tags for an image).

### CODE REVIEW

File: deepfence_server/reporters/scan/scan_result_actions.go [hunk 1]
```diff
@@ -61,6 +61,15 @@ func UpdateScanResultMasked(ctx context.Context, req *model.ScanResultsMaskReque
 		return err
 	}
 	defer tx.Close()
+	getImageStubQuery := func() string {
+		imageStubQuery := ""
+		if value == true {
```
Comment (noboruma) [inline deepfence_server/reporters/scan/scan_result_actions.go:66]: As indicated ;)

File: deepfence_server/reporters/scan/scan_result_actions.go [hunk 2]
```diff
@@ -61,6 +61,15 @@ func UpdateScanResultMasked(ctx context.Context, req *model.ScanResultsMaskReque
...
+			imageStubQuery = `MERGE (ist) -[ma:MASKED]-> (n) SET ma.masked = true`
+		} else {
+			imageStubQuery = `MATCH(ist) -[ma:MASKED] -> (n) DELETE ma`
+		}
+		return imageStubQuery
+	}
```
Comment (noboruma) [inline deepfence_server/reporters/scan/scan_result_actions.go:72]: Let's make this clearer by starting the query from:
```
MATCH (c) -[:IS] ->(ist)
```
Otherwise it will be hard to follow

File: deepfence_server/reporters/scan/scan_result_actions.go [hunk 3]
```diff
@@ -75,7 +84,12 @@ func UpdateScanResultMasked(ctx context.Context, req *model.ScanResultsMaskReque
         MATCH (s) -[:SCANNED] ->(e)
         MATCH (c:ContainerImage{node_id: e.docker_image_id}) -[:ALIAS] ->(t)
         MERGE (t) -[m:MASKED]->(n)
-        SET m.masked = $value`
+        SET m.masked = $value
+		WITH c,n
+		MATCH (c) -[:IS] ->(ist)
```
Comment (noboruma) [inline deepfence_server/reporters/scan/scan_result_actions.go:89]: Do we really need this line, can't we reuse `c` & `t`? or `r`?

**Review by noboruma — COMMENTED** [review]
LGTM just couple of questions around the new query

File: deepfence_server/reporters/scan/scan_result_actions.go (hunk 3 above)
Comment (varunsharma0286) [reply deepfence_server/reporters/scan/scan_result_actions.go:89]: I had to add this as the query was giving an error saying "WITH is required between SET and MATCH".

File: deepfence_server/reporters/scan/scan_result_actions.go (hunk 1 above)
Comment (varunsharma0286) [reply deepfence_server/reporters/scan/scan_result_actions.go]: Removed the local function and moved the code inline.

File: deepfence_server/reporters/scan/scan_result_actions.go (hunk 2 above)
Comment (varunsharma0286) [reply deepfence_server/reporters/scan/scan_result_actions.go]: Removed the local function and moved the code inline.

File: deepfence_server/reporters/scan/scan_result_actions.go (hunk 3 above)
Comment (noboruma) [reply deepfence_server/reporters/scan/scan_result_actions.go:80]: Yes but is `ist` no the same thing as `t`? (Asking not sure)
If so, we could simply do `WITH c,n,t`

Comment (varunsharma0286) [reply deepfence_server/reporters/scan/scan_result_actions.go:80]: `ist` and `t` are different nodes as below:
t -> ImageTag
ist -> ImageStub

File: deepfence_server/reporters/scan/scan_result_actions.go [hunk 4]
```diff
@@ -96,13 +108,37 @@ func UpdateScanResultMasked(ctx context.Context, req *model.ScanResultsMaskReque
...
+		WITH s, n, d
+		MATCH (s) -[:SCANNED]-> (c:ContainerImage) -[:ALIAS] ->(t) -[m:MASKED]-> (n)
+		WITH s, n, d, m, c
+		MATCH (c)-[:IS]->(ist)
+		SET d.masked=$value, m.masked=$value
+		WITH ist, n
+		%s`
+
+		imageStubQuery := ""
+		if value == true {
```
Comment (noboruma) [inline deepfence_server/reporters/scan/scan_result_actions.go:124]: Let fix the lint here

Comment (varunsharma0286) [reply deepfence_server/reporters/scan/scan_result_actions.go]: Done.

**Review by noboruma — APPROVED** [review]
